pkg/config/config.go

```diff
@@ -347,7 +361,14 @@ func load(fs *flag.FlagSet, args []string, loader func(string, bool, *Config) er
 		os.Exit(0)
 	}
 
-	if file == "" {
+	if features.Enabled(fs, featDynamicConfig) {
+		if dynamicConfigurationPath == "" {
+			return nil, fmt.Errorf("-config.dynamic-config-path flag required when using dynamic configuration")
+		} else if err := loader(dynamicConfigurationPath, configExpandEnv, &cfg); err != nil {
+			return nil, fmt.Errorf("error loading dynamic configuration file %s: %w", dynamicConfigurationPath, err)
+		}
```

Comment:
nit: hmm, I don't know if it should be _required_ when the feature flag is used. When it graduates out of being a feature flag, we wouldn't want it to be required then either. 

Can we make it optional, and let the user decide between normal/dynamic config?

Reply:
Do you mean to use the usage of the `--config.dynamic-config-path` as the implicit opt in?

Reply:
Yeah, i.e., still let people use `-config.file` even if the feature flag is enabled, but using `-config.dynamic-config-path` requires the feature flag to be set.

Reply:
Would the behavior in that scenario be that the normal file will be run through the template engine?

Reply:
IMO, I think there's two types of feature flags:

1. Enables optional new behavior (like `remote-configs`)
2. Switches out an implementation for a new one (like `integrations-next`)

I feel like this falls under the first bucket, where the intent is after graduating, it becomes an optional feature, so it might feel weird for users if during the feature flag phase, it's required when the feature is set. But it'd also be unlikely for someone to set the feature flag without already having the intent to use this functionality anyway. 

I don't really feel too strongly about this, and it's something we can fix if someone has a problem.

Reply:
Sounds reasonable, if it becomes an issue more than happy to look at the feedback and change accordingly.